core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -1888,7 +1899,10 @@ protected String getEffectiveSearchBase(boolean shared) {
                                              boolean shared) throws UserStoreException {
         boolean debug = log.isDebugEnabled();
         List<String> list = new ArrayList<String>();
-		/*
+        if (userName == null) {
+            throw new UserStoreException("userName value is null.");
+        }
```

Comment:
moved to top. This username null check is done mainly to handle the NPE in userCache ConcurrentHashMaps.